src/PoolManager.sol

```diff
@@ -175,25 +179,38 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
     }
 
     modifier onlyByLocker() {
-        address locker = Lockers.getCurrentLocker();
-        if (msg.sender != locker) revert LockedBy(locker);
+        _checkLocker(msg.sender, Lockers.getCurrentLocker(), Lockers.getCurrentHook());
         _;
     }
 
+    function _checkLocker(address caller, address locker, IHooks hook) internal pure {
```

Comment:
since the modifier is the only calsite, why have as a separate internal fn?

Reply:
stack too deep :/ I could remove it but then I can't save msg.sender or Lockers.getCurrentLocker or Lockers.getCurrentHook